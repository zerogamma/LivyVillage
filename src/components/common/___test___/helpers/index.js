import React from 'react'
import renderer from 'react-test-renderer'
import theme from '../../../../utils/style/theme'
import { ThemeProvider } from 'styled-components'

export function renderWithTheme(component) {
    return renderer.create(
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
    )
}
