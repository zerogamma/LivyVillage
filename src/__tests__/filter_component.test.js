import React from "react"
import Filter from "../components/village/filter";

import 'jest-styled-components';
import { renderWithTheme } from '../components/common/___test___/helpers';

describe("Filter Component render test", () => {
    it("renders Filter component correctly", () => {
      const data = {
            'age':'166',
            'weight':'35',
            'height':'106',
            'hair':'Red',
            'profession':[
                "Cook",
                "Baker",
                "Miner"
            ]};
      const comp = renderWithTheme(<Filter currentFilter={data} filter={()=>{}} />).toJSON()
      expect(comp).toMatchSnapshot()
    })
  })