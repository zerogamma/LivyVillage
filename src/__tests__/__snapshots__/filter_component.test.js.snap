// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component render test renders Filter component correctly 1`] = `
.c0 {
  -webkit-transition: width 1s ease-out;
  transition: width 1s ease-out;
  height: 100%;
  background-color: lightgray;
  min-height: 1000px;
  width: 45px;
}

.c3 {
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
  position: relative;
  top: 50%;
}

.c1 {
  top: 15px;
  position: relative;
  text-align: end;
  margin-right: 25px;
}

.c2 {
  position: fixed;
  left: 10px;
}

@media (max-width:767px) {
  .c3 {
    top: 4%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root c2"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
      />
    </svg>
  </div>
  <div
    className="c3"
  >
    <div
      className=""
    >
      Filter
    </div>
  </div>
</div>
`;
